docs/architecture/process-view.md

```diff
@@ -0,0 +1,80 @@
+# Process View
+
+* Solution architect
+
+
+
+* Sequence
+* Communication
+* Activity
+* Timing
+* Interaction overview
+
+
+
+<figure><img src="../.gitbook/assets/image.png" alt=""><figcaption><p><a href="https://docs.axoniq.io/reference-guide/architecture-overview">https://docs.axoniq.io/reference-guide/architecture-overview</a></p></figcaption></figure>
+
+
+
+Micro-service main component communication
+
+* command handling
+
+```mermaid
+sequenceDiagram
+    actor u as User
+    participant c as Controller
+    participant a as Aggregate
+    participant es as Event Store / Message Queue
+    participant eh as Event Handlers
+
+    u ->> c: Command with modification request
+    c ->> a: Command routed via Command Gateway
+    a ->> es: Fetch all the events with given aggregate id
+    es ->> a: Recreate the aggregate state based on those events
+    a ->> a: Handle provided command, it may generate events
+    a ->> a: Handle new events in appropriate event sourcing handlers (modify aggregate state)
+    a ->> es: Store generated valid events
+    es ->> eh: Pass events to other event handlers
+    a ->> c: No response (Async) or return generated aggregate identifier
+    c ->> u: Return result
+```
+
+* query service updating the state
+
+```mermaid
+sequenceDiagram
+    participant es as Event Store / Message Queue
+    participant qs as Queue Service
+    participant ts as Token Store
+    participant p as Projector
+    participant r as Repository
+
+    es ->> qs: Service has event handlers for given type
+    qs ->> ts: This instance is responsible for handling event with that aggregate id
+    qs ->> p: Pass events to all the registered event handlers
```

Comment:
"to all event handlers", but the diagram goes to a Projector

Reply:
Projector contains logic for updating and retrieving a view

For example, here - projector is responsible for updating user view and responsible for handling user queries
```kotlin
@Service
class UserProjector(
	private val repository: UserViewRepository,
) {

	@EventHandler
	fun on(event: UserCreatedEvent) {
		val view = UserView(
			id = event.userId,
			name = event.name,
			email = event.email,
		)
		repository.save(view)
	}

	@QueryHandler
	fun handle(query: FindUserByIdQuery): UserView =
		repository.findById(query.userId) ?: throw UserNotFoundException(query.userId)
}
```

You could define multiple event handlers and multiple query handlers
but here projector aggregates handlers that touch a single view

There can be multiple different projectors handling the same events